### PR TITLE
SCB-909 Add fault tolerance for service comb TCC  (WIP)

### PR DESCRIPTION
**Pull request by cherrylzhao**
Add Tcc LoaderBalanceSender & refactor omega tolerance mechanics

### CODE REVIEW

File: omega/omega-connector/omega-connector-grpc/src/main/java/org/apache/servicecomb/saga/omega/connector/grpc/tcc/GrpcCoordinateStreamObserver.java [hunk 1]
```diff
@@ -6,33 +6,34 @@
  * (the "License"); you may not use this file except in compliance with
```
Comment (WillemJiang) [inline omega/omega-connector/omega-connector-grpc/src/main/java/org/apache/servicecomb/saga/omega/connector/grpc/tcc/GrpcCoordinateStreamObserver.java]: Hi, I think You need to change the template.

File: omega/omega-connector/omega-connector-grpc/src/test/java/org/apache/servicecomb/saga/omega/connector/grpc/tcc/LoadBalanceContextBuilderTest.java [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+  @Test(expected = IllegalArgumentException.class)
+  public void throwExceptionWhenAddressIsNotExist() {
+    when(clusterConfig.getAddresses()).thenReturn(new ArrayList<String>());
+    tccLoadBalanceContextBuilder.build();
+  }
+
+  @Test
+  public void buildSagaLoadBalanceContextWithoutSsl() {
+
```
Comment (WillemJiang) [inline omega/omega-connector/omega-connector-grpc/src/test/java/org/apache/servicecomb/saga/omega/connector/grpc/tcc/LoadBalanceContextBuilderTest.java]: Do we miss some test code here?

Comment (cherrylzhao) [reply omega/omega-connector/omega-connector-grpc/src/test/java/org/apache/servicecomb/saga/omega/connector/grpc/tcc/LoadBalanceContextBuilderTest.java]: I'll add these case when integrated with saga implement

File: omega/omega-connector/omega-connector-grpc/src/main/java/org/apache/servicecomb/saga/omega/connector/grpc/tcc/GrpcCoordinateStreamObserver.java (hunk 1 above)
Comment (cherrylzhao) [reply omega/omega-connector/omega-connector-grpc/src/main/java/org/apache/servicecomb/saga/omega/connector/grpc/tcc/GrpcCoordinateStreamObserver.java]: sorry, I haven't understand what it mean.

**Comment by WillemJiang** [comment]
@cherrylzhao  as the CI passed, I merge it first and polish the code later.

**Pull request merged**